alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/EventScanner.java

```diff
@@ -75,11 +72,11 @@ private void pollEvents() {
   }
 
   private void saveUncompensatedEventsToCommands() {
-    eventRepository.findFirstUncompensatedEventByIdGreaterThan(nextEndedEventId, TxEndedEvent.name())
+    eventRepository.findByTypeAndIdGreaterThan(nextAbortedEventId, TxAbortedEvent.name())
```

Comment:
we can't fully depend on AbortedEvent here, since TxStartedEvent and TxEndedEvent of some sub transaction may arrive later, causing some events are not compensated.